### PR TITLE
[SPARK-34642][DOCS][ML] Fix TypeError in Pyspark Linear Regression docs

### PR DESCRIPTION
**Pull request by srowen**
### What changes were proposed in this pull request?

Fix a call to setParams in the Linear Regression docs example in Pyspark to avoid a TypeError.

### Why are the changes needed?

The example is slightly wrong and we should not show an error in the docs.

### Does this PR introduce _any_ user-facing change?

None

### How was this patch tested?

Existing tests

### CODE REVIEW

File: python/pyspark/ml/regression.py [hunk 1]
```diff
@@ -180,10 +180,7 @@ class LinearRegression(_JavaRegressor, _LinearRegressionParams, JavaMLWritable,
     >>> test1 = spark.createDataFrame([(Vectors.sparse(1, [0], [1.0]),)], ["features"])
     >>> abs(model.transform(test1).head().newPrediction - 1.0) < 0.001
     True
-    >>> lr.setParams("vector")
-    Traceback (most recent call last):
-        ...
-    TypeError: Method setParams forces keyword arguments.
+    >>> lr.setParams(featuresCol="vector")
```
Comment (srowen) [inline python/pyspark/ml/regression.py:183]: @HyukjinKwon is it correct here just to specify no output from this command? I don't think we care about testing any output of this. I don't know how python docs work though :)

Comment (dongjoon-hyun) [reply python/pyspark/ml/regression.py:183]: Hi, @srowen . +1 for removing this ancient `TypeError` example (added 5 year ago). In this case, we can do partial check like the following.
```python
>>> lr.setParams(featuresCol="vector")
LinearRegression...
```

Comment (HyukjinKwon) [reply python/pyspark/ml/regression.py:183]: Yeah ^^ seems correct

**Review by HyukjinKwon — APPROVED** [review]

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM. Merged to master/3.1.

**Pull request closed without merging**